### DISCUSSION TITLE
Replace yii2tech/ar-softdelete package

### DISCUSSION THREAD

**Discussion by bencroker** [craftcms/cms#10435] [Ideas] [answered]
The [yii2tech/ar-softdelete](https://github.com/yii2tech/ar-softdelete) package, which Craft requires for soft delete functionality, has been abandoned with the following explanation: https://www.patreon.com/posts/60968716

The author states:

> If you are in a dire need for abandoned package advancement, you should create your own fork of it and continue development on your own.

This seems like a good time to either fork the package, or bring the `SoftDeleteBehavior` and `SoftDeleteQueryBehavior` classes into Craft core.

**Comment by brandonkelly** [comment, marked as answer]
The package is stable and does its job so I’m not too concerned about it, beyond the awkward Composer notice and possibility of the repo getting deleted. So we decided it’s best to just bundle the library as-is directly into Craft. (b1df8ffed60c72c3b73ca88951822f52344e9e72)
